secure/storage/src/tests/vault.rs

```diff
@@ -6,42 +6,174 @@ use crate::{
+        U64_VALUE_1
     );
-    let mut storage1 = VaultStorage::new(
-        VAULT_HOST.into(),
-        VAULT_ROOT_TOKEN.into(),
-        Some("test1".into()),
+    assert!(storage_2.get(U64_KEY_1).is_err());
+    assert!(storage_3.get(U64_KEY_1).is_err());
+
+    // Verify create and sets are unique across namespaces
+    storage_2
+        .create(U64_KEY_2, Value::U64(U64_VALUE_2), &Policy::public())
+        .unwrap();
+    storage_2.set(U64_KEY_2, Value::U64(U64_VALUE_2)).unwrap();
+    assert_eq!(
+        storage_2.get(U64_KEY_2).unwrap().value.u64().unwrap(),
+        U64_VALUE_2
     );
+    assert!(storage_1.get(U64_KEY_2).is_err());
+    assert!(storage_3.get(U64_KEY_2).is_err());
+
+    // Verify create_if_not_exists are unique across namespaces
+    storage_3
+        .create_if_not_exists(U64_KEY_3, Value::U64(U64_VALUE_3), &Policy::public())
+        .unwrap();
+    assert_eq!(
+        storage_3.get(U64_KEY_3).unwrap().value.u64().unwrap(),
+        U64_VALUE_3
+    );
+    assert!(storage_1.get(U64_KEY_3).is_err());
+    assert!(storage_2.get(U64_KEY_3).is_err());
+}
+
+/// Runs the test suite on a VaultStorage instance that does not use distinct namespaces
+fn test_suite_no_namespaces() {
+    let mut storage = create_initialized_vault_storage(None);
+    suite::execute_all_storage_tests(&mut storage);
+}
+
+/// Runs the test suite on a VaultStorage instance that supports multiple distinct namespaces.
+/// Tests should be able to run across namespaces without interfering.
+fn test_suite_multiple_namespaces() {
+    let mut storage_1 = create_initialized_vault_storage(Some(VAULT_NAMESPACE_1.into()));
+    let mut storage_2 = create_initialized_vault_storage(Some(VAULT_NAMESPACE_2.into()));
+
+    suite::execute_all_storage_tests(&mut storage_1);
+    suite::execute_all_storage_tests(&mut storage_2);
+}
+
+/// Creates and initializes a VaultStorage instance for testing. If a namespace is specified, the
+/// instance will perform all storage operations under that namespace.
+fn create_initialized_vault_storage(namespace: Option<String>) -> VaultStorage {
+    let mut storage = VaultStorage::new(VAULT_HOST.into(), VAULT_ROOT_TOKEN.into(), namespace);
+    storage
+        .reset_and_clear()
+        .expect("Failed to reset VaultStorage after creation!");
 
-    test_vault(&mut storage0);
-    test_vault(&mut storage1);
-    suite::execute_all_storage_tests(&mut storage0);
-    suite::execute_all_storage_tests(&mut storage1);
-    storage.reset_and_clear().unwrap();
+    initialize_vault_storage(&mut storage);
+    storage
 }
 
-/// Creates and returns a new VaultStorage instance for testing purposes.
-pub fn test_vault(storage: &mut VaultStorage) {
+/// Initializes default test policies for a given VaultStorage instance and checks the instance is
+/// ready to be used (e.g., by ensuring subsequent read and write operations complete successfully).
+fn initialize_vault_storage(storage: &mut VaultStorage) {
```

Comment:
This is your basic test that would only pass if we have namespace separation, similarly the battery of tests in suite would do the same

I would also call this run_vault_test_suite and call it directly in the test method

Reply:
Interesting.. thanks! I wasn't sure you originally wanted to model this a separate test, but I guess that makes sense too! It's now called 'test_vault_tokens_and_basic_ops()' so as to not interfere with the tests that actually call the test suite.